firestore/runtime/src/main/java/io/quarkiverse/googlecloudservices/firestore/runtime/FirestoreProducer.java

```diff
@@ -32,10 +35,37 @@ public class FirestoreProducer {
     @Default
     public Firestore firestore() throws IOException {
         GcpBootstrapConfiguration gcpConfiguration = gcpConfigHolder.getBootstrapConfig();
-        FirestoreOptions.Builder builder = FirestoreOptions.newBuilder().setCredentials(googleCredentials)
+        FirestoreOptions.Builder builder = FirestoreOptions.newBuilder()
+                .setCredentials(googleCredentials)
                 .setProjectId(gcpConfiguration.projectId.orElse(null));
         firestoreConfiguration.hostOverride.ifPresent(builder::setHost);
-
+        builder.setRetrySettings(buildRetrySettings());
         return builder.build().getService();
     }
+
+    private RetrySettings buildRetrySettings() {
+        RetrySettings.Builder retrySettingsBuilder = RetrySettings.newBuilder();
+        firestoreConfiguration.retry.ifPresent(retryConfiguration -> {
```

Comment:
I would move this `ifPresent` out of the `buildRetrySettings` method, to avoid building a retrySetting if no retry is configured

Reply:
Good idea!